### PR TITLE
fix(gatsby-plugin-sharp): return correct height

### LINKED ISSUE #25149: gatsby-plugin-sharp fluid image not returning proper sizes with given both maxHeight and maxWidth
## Description

I am creating a portfolio for high resolution pictures and would like to optimize the images that are produced by the sharp plugin. When I provide both maxWidth and maxHeight to my graphql query, images that are returned are considerably smaller than the original image provided. 

### Steps to reproduce

I have a minimal reproduction [here ](https://github.com/imakida/gatsby_bug_repro)

The main issue lies in my graphql query 
```
export const query = graphql`
  query {
    allFile(filter: {relativePath: {regex:"/Gallery/"}})
    {
      edges{
        node{
          id, childImageSharp{
                     fluid(maxWidth:3840, maxHeight:2160, quality:72 fit:INSIDE){
              ...GatsbyImageSharpFluid_withWebp
            }
          }
        }
      }
    }
  }
`
```
### Expected result

Images should be as large as possible without exceeding the original size. Images should be constrained to the provided minWidth and maxHeight options. 

For example: an image with a native size of 720 x 1080 should be as tall as an image that is 1920 x 1080 if the maxWidth > 720 and maxHeight > 1080. 

### Actual result

I have an image that is 720  x 1080 and that gets sized to 270 x 405. 

### Environment

System:
    OS: Linux 4.4 Ubuntu 18.04.4 LTS (Bionic Beaver)
    CPU: (8) x64 Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
    Shell: 4.4.20 - /bin/bash
  Binaries:
    Node: 10.19.0 - /usr/bin/node
    Yarn: 1.22.4 - /usr/bin/yarn
    npm: 6.14.5 - /usr/bin/npm
  Languages:
    Python: 2.7.17 - /usr/bin/python
  npmPackages:
    gatsby: ^2.23.3 => 2.23.3
    gatsby-image: ^2.4.7 => 2.4.7
    gatsby-plugin-manifest: ^2.4.11 => 2.4.11
    gatsby-plugin-offline: ^3.2.9 => 3.2.9
    gatsby-plugin-react-helmet: ^3.3.4 => 3.3.4
    gatsby-plugin-sharp: ^2.6.11 => 2.6.11
    gatsby-source-filesystem: ^2.3.11 => 2.3.11
    gatsby-transformer-sharp: ^2.5.5 => 2.5.5
  npmGlobalPackages:
    gatsby-cli: 2.12.44
Annotation 2020-06-19 213827

### PR DESCRIPTION
**Pull request by imakida**
When both a maxWidth and maxHeight option is provided, the height of the
image should be determined by the height and width of the image if the
image is smaller than the provided maxWidth and maxHeight options. i.e.
if the maxWidth and maxHeight option is 3840 and 2160, but the image is
720 x 1080, the maxHeight should be 1080.

fixes #25149

### CODE REVIEW

File: packages/gatsby-plugin-sharp/src/index.js [hunk 1]
```diff
@@ -543,7 +543,7 @@ async function fluid({ file, args = {}, reporter, cache }) {
     }
 
     if (options.maxWidth !== undefined && options.maxHeight !== undefined) {
-      arrrgs.height = Math.round(size * (options.maxHeight / options.maxWidth))
+      arrrgs.height = Math.round(size * (maxHeight / maxWidth))
     }
```
[the remark this answers is not in the document]
Comment (imakida) [reply packages/gatsby-plugin-sharp/src/index.js:547]: Thanks @polarathene for taking a look at this and helping me through my first PR!

Shouldn't `fixed` images with both a specified `width` and `height` be exactly _width x height_ when possible (native width and height > `width` and `height`)?

I mocked this up without changing anything. Here's the graphql query:
```
export const query = graphql`
  query {
    allFile(filter: {relativePath: {regex:"/Gallery/"}})
    {
      edges{
        node{
          id, childImageSharp{
              fixed(width:1920, height:1080, quality:72){
                ...GatsbyImageSharpFixed_withWebp
            }
          }
        }
      }
    }
  }
`
```

Images greater than or equal to get cropped to 1920 x 1080 per the graphql query. The 720 x 1080 image becomes 720 x 405 to honor the aspect ratio of the given `width` and `height` (1920 and 1080). The 720 x 1080 image does not grow larger than its native size. I believe this should be the expected behavior for `fixed` images with a give `width` and `height`. 

image

[the remark this answers is not in the document]
Comment (polarathene) [reply packages/gatsby-plugin-sharp/src/index.js:547]: @imakida right you are! :grimacing: 

Sorry for that, I probably should do code reviews so late at night, I was under the impression the line referenced `options.maxHeight` too, and just overall wasn't thinking particularly well :sweat_smile:

**Comment by polarathene** [comment]
@imakida tests are failing, you'll need to update the snapshots.

eg [This sharp test failing due to height values differing from current snapshot](https://app.circleci.com/pipelines/github/gatsbyjs/gatsby/44874/workflows/fcba7c65-d367-4ed4-af62-2c3ec3dbc849/jobs/455595/steps):

```
 gatsby-plugin-sharp › fluid › calculate height based on width when maxWidth & maxHeight are present
    expect(received).toMatchSnapshot()
    Snapshot name: `gatsby-plugin-sharp fluid calculate height based on width when maxWidth & maxHeight are present 5`

packages/gatsby-plugin-sharp/src/__tests__/index.js:289:60
 ```

**Comment by imakida** [comment]
@polarathene I made changes to correct the issue with the default `fit: COVER` option for `fluid` and updated the Jest snapshot for the `fit: INSIDE` option.  Tests are passing now.

**Comment by polarathene** [comment]
@wardpeet Please merge? :)

**Pull request merged**